### PR TITLE
Revert "Revert 'General Update'"

### PR DESCRIPTION
**Pull request by jphuse**
This reverts commit aa68e8fe

- Starting to look into BannedWords not registering
- Fixed RandomTP formatting
- Fixed added configurability to Ranks command (Need to implement into Rankup)
- Fixed permission error with DCSync
- Remove staff tagging notifications when server staffless

### CODE REVIEW

**Comment by jphuse** [comment]
Fixed BannedWordsListener.java

**Comment by jphuse** [comment]
Updated BannedWords to check for usernames and displaynames, removed kill listener.

File: src/com/Jacksonnn/DCCore/BannedWords/BannedWordsListener.java [hunk 1]
```diff
@@ -2,25 +2,30 @@
...
+        bannedWords = ConfigManager.bannedWords.get().getStringList("AntiCurse.bannedWords");
+
+        for (String bannedWord : bannedWords) {
+            if (event.getMessage().toUpperCase().contains(bannedWord.toUpperCase()) || player.getName().toUpperCase().contains(bannedWord.toUpperCase()) || player.getDisplayName().toUpperCase().contains(bannedWord.toUpperCase())) {
+                if (player.hasPermission("DCCore.AntiCurse.bypass")) {
+                    return;
+                } else {
+                    event.setCancelled(true);
+                    event.getPlayer().sendMessage(GeneralMethods.prefix + " Please rethink your choice of words... (check your username, nickname, or chat message!!!)");
+                    event.getPlayer().sendMessage(ChatColor.YELLOW + "Banned word used: " + bannedWord);
```
Comment (Celtican) [inline src/com/Jacksonnn/DCCore/BannedWords/BannedWordsListener.java]: If a player sends a message with multiple banned words, they receive the "Please rethink your choice of words..." message multiple times. Consider naming a String[] variable (initialize it to null) before entering the loop of bannedWords. If a bannedWord is detected, if the String[] is not initialized, initialize it, then add the word to the String[]. At the end of the loop, if String[] is not null, send the message and cancel the event. Also, consider testing for the DCCore.AntiCurse.bypass permission outside the loop
```suggestion
        if (player.hasPermission("DCCore.AntiCurse.bypass"))
           return;
        ArrayList<String> detectedWords = null;
        for (String bannedWord : bannedWords) {
            if (message.contains(bannedWord)) {
                if (detectedWords == null)
                   detectedWords = ArrayList<>();
                 detectedWords.add(bannedWord);
              }
         }
         if (detectedWords != null) {
            event.setCancelled(true);
            event.getPlayer().sendMessage(GeneralMethods.prefix + " Please rethink your choice of words... (check your username, nickname, or chat message!!!)");
            StringBuilder message = new StringBuilder(ChatColor.YELLOW + "Banned words used: ");
            for (String word : detectedWords)
               message.append(word).append(" ");
            event.getPlayer().sendMessage(message);
         }
```

**Review by Celtican — COMMENTED** [review]
See comment on lines +21 to +28 in BannedWordsListener for notes on bug

**Review by Celtican — APPROVED** [review]

**Pull request merged**